tests/src/Kernel/Entity/GetUserGroupsTest.php

```diff
@@ -234,6 +228,32 @@ public function testIsMemberStates() {
     $this->assertFalse(Og::isMemberPending($this->group1, $this->user3));
   }
 
+  /**
+   * Tests retrieval of groups filtered by roles.
+   */
+  public function testGetGroupsByRoles() {
+    /** @var \Drupal\og\MembershipManagerInterface $membership_manager */
+    $membership_manager = \Drupal::service('og.membership_manager');
```

Comment:
I would say to store the membership manager in a property and populate it in the setup using `$this->membershipManager = $this->container->get('og.membership_manager');` but I see this is actually consistent with the other tests in this class. So it is fine as it is right now, but if you want to improve it feel free!